### ISSUE TITLE
External checks across namespaces seem broken in k8s 1.22

### ISSUE THREAD

**Issue by jbeda** [kuberhealthy/kuberhealthy#975]
**Describe the bug**
External checks seem to fail when in non-kuberhealthy namespace on newer k8s versions

**Steps To Reproduce**
Create the `dns-status-external.yaml` in the `default` namespace

**Expected behavior**
The check should succeed (if the cluster is healthy)

**Screenshots**
We get an error that looks something like this:
```
    "default/dns-status-external": {
      "OK": false,
      "Errors": [
        "Check execution error: pod deleted unexpectedly"
      ],
      "RunDuration": "",
      "Namespace": "default",
      "LastRun": "2021-07-09T21:19:50.035838028Z",
      "AuthoritativePod": "kuberhealthy-67d5fb67b-c6njn",
      "uuid": "77b211b6-9b11-426d-803c-c36c606cd028"
    },
```

**Versions**
 - Cluster OS: whatever kind users by default
 - Kubernetes Version: 1.22.1
 - Kuberhealthy Release or build: 2.5.0

**Additional context**
It looks like, in newer versions of k8s, you aren't allowed to have owner refs that cross namespace boundaries. When I launched the pod manually (would be great to have a way to capture yaml that kuberhealthy uses to launch pods) I was able to reproduce the problem. Removing the ownerRef seemed to fix it.

**Comment by integrii** [comment]
Hey Joe!  We've met up a couple times in Seattle and Bellevue, and I am very happy to cross paths again.  I really appreciate you selecting Kuberhealthy for review on your TGIK stream today and extra appreciate you taking the time to file a bug here!

This bug is fixed, but hasn't made it into a release yet... We fixed this one in #977 by simply skipping the ownerref when the checker pod is being created outside of the namespace where kuberhealthy lives.  I am sad to see this feature stop working, because it was very helpful in keeping Kuberhealthy checker pods cleaned up.  I expressed this to the relevant parties in the appropriate SIG, but I think namespaces are being more strongly considered a security boundary as Kubernetes moves forward.

If I can do anything for you, please don't hesitate to reach out.

I will leave this issue open until we do another release.

**Comment by jbeda** [comment]
Love the project! Thanks for the update and good to know it is already solved.

**Comment by integrii** [comment]
That means a ton coming from you!  I am closing this issue because this fix is now released in [Kuberhealthy 2.6.0](https://github.com/kuberhealthy/kuberhealthy/releases/tag/v2.6.0).
